io-sim-classes/src/Control/Monad/Class/MonadSTM/Strict.hs

```diff
@@ -100,7 +110,11 @@ toLazyTVar :: StrictTVar m a -> LazyTVar m a
 toLazyTVar StrictTVar { tvar } = tvar
 
 newTVar :: MonadSTM m => a -> STM m (StrictTVar m a)
+#if CHECK_TVAR_INVARIANT
```

Comment:
How about `newTVar a = newTVarWithInvariantIO  (const Nothing)`?

Reply:
That would not compile the lhs is in `STM m` while the right hand side is in monad `m`.

Reply:
Ah, right. And I suppose there's no `newTVarWithInvariant` lying around.

I wonder if a little more refactoring could achieve something similar and thereby avoid another `#if`. But I'll leave it there; I'm pretty sure you get the point of the question and I know you can judge it 👍